core/drivers/crypto/caam/acipher/caam_prime.c

```diff
@@ -663,7 +663,7 @@ static void do_checks_primes(uint32_t *desc, const struct caambuf *p,
 	 * We started with 128, 192, or 256 bytes in the OFIFO before we moved
 	 * check_len bytes into MATH registers.
 	 */
-	if (p->length > 128 + check_len) {
+	if ((int)p->length > 128 + check_len) {
```

Comment:
Given that `p->length` is `size_t`, which is effectively `unsigned long`, wouldn't it be better to cast `check_len` to `size_t`? My concern is if `p->length` is large enough it will cast to a negative `int`. But perhaps it can't happen here?